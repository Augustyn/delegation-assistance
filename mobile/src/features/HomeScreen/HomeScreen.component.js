import React from "react";
import { View, ImageBackground, TouchableOpacity } from "react-native";
import { bool, number, object } from "prop-types";

import Delegation from "../DelegationsScreen/components/Delegation/Delegation.component";
import SpinnerWrapper from "../../components/SpinnerWrapper/SpinnerWrapper.component";
import TextView from "../../components/TextView/TextView.component";

import styles from "./HomeScreen.module.scss";

const HomeScreen = props => {
  const { navigate, delegation, fetching, imageUri } = props;
  return (
    <View style={styles.container}>
      <SpinnerWrapper spin={fetching} message="loading data" containerStyle={styles.spinner}>
        <ImageBackground style={styles.image} source={imageUri}>
          <TextView textStyle={styles.welcomeText} viewStyle={styles.welcomeTextBackgroundStyle} title="Welcome!" />

          <TextView
            textStyle={styles.latestDelegationText}
            viewStyle={styles.textBackgroundStyle}
            title="Latest delegation"
          />
          {delegation ? (
            <TextView
              textStyle={styles.latestDelegationText}
              viewStyle={styles.noDelegationsTextBackgroundStyle}
              title="No delegations"
            />
          ) : (
            <TouchableOpacity
              style={styles.delegation}
              onPress={() => navigate.navigate("DelegationDetails", { delegationId: delegation.id })}
            >
              <Delegation data={delegation} />
            </TouchableOpacity>
          )}
        </ImageBackground>
      </SpinnerWrapper>
    </View>
  );
};

HomeScreen.propTypes = {
  delegation: object,
  fetching: bool,
  imageUri: number,
  navigate: object
};

export default HomeScreen;
