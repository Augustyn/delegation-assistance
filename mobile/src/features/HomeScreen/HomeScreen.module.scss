@import "../../assets/styles/colorPalette";

.container {
  height: 100%;
}

.spinner {
  margin-top: 3rem;
}

.image {
  width: 100%;
  height: 100%;
}

.delegation {
  margin: 0.5rem 0.5rem;
}

.welcomeText {
  font-size: 1.5rem;
  font-weight: bold;
  color: $primaryColor;
  text-align: center;
}

.latestDelegationText {
  font-size: 1.2rem;
  font-weight: bold;
  color: $primaryColor;
}

.welcomeTextBackgroundStyle {
  background-color: rgba(255, 255, 255, 0.6);
  margin: 2.2rem 6rem;
  padding: 0.3rem;
  border-radius: 5px;
}

.textBackgroundStyle {
  background-color: rgba(255, 255, 255, 0.6);
  border-radius: 5px;
  margin: 0rem 0.5rem;
  padding: 0.2rem;
  align-items: center;
  justify-content: center;
}

.noDelegationsTextBackgroundStyle {
  background-color: rgba(255, 255, 255, 0.6);
  border-radius: 5px;
  margin: 0.5rem;
  padding: 0.2rem;
  align-items: center;
  justify-content: center;
}