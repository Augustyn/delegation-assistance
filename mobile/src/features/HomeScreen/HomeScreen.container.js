import React, { Component } from "react";
import { connect } from "react-redux";

import HomeScreen from "./HomeScreen.component";

class HomeScreenContainer extends Component {
  static navigationOptions = {
    title: "Home"
  };

  render() {
    return <HomeScreen />;
  }
}

export default connect(
  null,
  null
)(HomeScreenContainer);
