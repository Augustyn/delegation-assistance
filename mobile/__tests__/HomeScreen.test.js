import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import HomeScreen from "../src/features/HomeScreen/HomeScreen.component";

describe("Home snapshot", () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it("renders the Home screen", async () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
