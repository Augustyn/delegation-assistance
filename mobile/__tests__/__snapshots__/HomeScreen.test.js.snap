// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders the Home screen 1`] = `
<View>
  <Text>
    Home Screen
  </Text>
</View>
`;
